ivys.js

```diff
@@ -26,14 +26,55 @@ IvyApp.prototype.removeItem = function() {
     
 };
 
-IvyApp.prototype.addTag = function() {
-};
-
-IvyApp.prototype.removeTag = function() {
-};
-
 // Footer menu arrow function 
 IvyApp.prototype.toggleFooterMenu = function() {
     // Toggles footer menu
     $(this).parent().parent().children('.menu-footer').collapse('toggle');
 }
+
+
+// Product State and Product Class skeletons
+
+var emptyProductState = function(params) {
+    
+    return {
+        // seaWinds Id, SKU number?
+        id: params.seaWindsId,
+        // Product name
+        name: params.name,
+        // product tags
+        tags: [],
```

Comment:
I think the number tags will be small enough not to matter, but think about the kinds of operations you're going to want to do on tags and the performance characteristics of the datatype you choose.